docs/src/main/paradox/server-side/graceful-termination.md

```diff
@@ -0,0 +1,75 @@
+# Graceful termination
+
+## Graceful termination using `ServerTerminator`
+
+Akka HTTP provides two APIs to "stop" the server, either of them are available via the
+@java[@javadoc[ServerBinding](akka.http.javadsl.ServerBinding)]
+@scala[@scaladoc[ServerBinding](akka.http.scaladsl.Http$$ServerBinding)]
+obtained from starting the server (by using any of the `bind...` methods on the
+@java[@javadoc[Http](akka.http.javadsl.Http)]@scala[@scaladoc[Http](akka.http.scaladsl.HttpExt)] extension).
+
+The first method, called `unbind()` causes the server to *stop accepting new connections*, however any existing
+connections that are still being used will remain active.
+It only unbinds the port on which the http server has been listening. This allows applications to finish streaming any
+responses that might be still in flight and eventually terminate the entire system. If your application uses long-lived
+connections, this does mean that these can delay the termination of your system indefinitely.
+
+A better and more graceful solution to terminate an Akka HTTP server is to use the
+@java[@javadoc[ServerBinding.terminate(Duration)](akka.http.javadsl.ServerBinding#terminate-java.time.Duration-)]
+@scala[@scaladoc[ServerBinding.terminate(FiniteDuration)](akka.http.scaladsl.Http$$ServerBinding#terminate%28FiniteDuration%29:Future[HttpTerminated])]
+method, which not only performs the unbinding, but also
+handles replying to new incoming requests with (configurable) "terminating" HTTP responses.
+It also allows setting a deadline after which any connections that are still alive will be shut down forcefully.
+More precisely, termination works by following these steps:
+
+Step 1, the server port is unbound and no new connections will be accepted (same as invoking `unbind()`).
```

Comment:
When reading through the steps it seems to me they are in fact not at all steps but just different aspects of the termination, so maybe skip the "Step N," part?

Reply:
Time wise they are executed in such order... so while not explicit steps kind of in this order hm... should I remove `step`? In scaladoc it was like: 

```
   * 2) Handle in-flight request:
   * - if a request is "in-flight" (being handled by user code), it is given `hardDeadline` time to complete,
```

Reply:
Step one is a step, and then 2 but 3 is just what happens as soon as step 1 has, and same for 4 I think. Felt it was somewhat confusing.

Reply:
Ok, I'll remove the steps